shell/platform/linux/fl_texture_registrar_test.cc

```diff
@@ -104,3 +104,24 @@ TEST(FlTextureRegistrarTest, MarkTextureFrameAvailable) {
   EXPECT_TRUE(
       fl_texture_registrar_mark_texture_frame_available(registrar, texture));
 }
+
+// Test that mutex should be unlocked after calling register_texture.
+TEST(FlTextureRegistrarTest, RegistrarRegisterTextureMutex) {
```

Comment:
I would drop this test - the mutex is not an external detail of this object. To do this test we've added a method that's only used for testing. @cbracken WDYT.

Reply:
I've added a function called `fl_texture_registrar_get_textures_locked` in this PR to test whether the mutex is locked or not. I don't know whether it can be the method that's only used for testing.

Reply:
That's the one I'm referring to. The only way this test is going to fail is if the clear mutex unlock code is removed, which seems like an unlikely thing to occur. I think a better test would be to access the object with multiple threads as that's the failure case we have now and don't want to regress to. The test wouldn't always fail but it might fail sometimes which would indicate a regression.